### ISSUE TITLE
Update setup example to use Skylark http_archive

### ISSUE THREAD

**Issue by ash2k** [bazelbuild/rules_go#1468]
[Bazel 0.12.0 release notes](https://blog.bazel.build/2018/04/11/bazel-0.12.html) say:
> The native http_archive rule has been deprecated. Use the Skylark version available via load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive") instead.
>
> The native git_repository rule has been deprecated. Use the Skylark version available via load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository") instead.

[Documentation for native `git_repository`](https://docs.bazel.build/versions/master/be/workspace.html#git_repository) says (follow the link for the details):
> Warning: this rule has several limitations. We recommend using http_archive instead for more robust and secure performance.

Example should be something like this:
```bzl
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "io_bazel_rules_go",
    sha256 = "f70c35a8c779bb92f7521ecb5a1c6604e9c3edd431e50b6376d7497abc8ad3c1",
    url = "https://github.com/bazelbuild/rules_go/releases/download/0.11.0/rules_go-0.11.0.tar.gz",
)
```

The setup example for a particular commit can still use `http_archive` like this:
```bzl
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "io_bazel_rules_go",
    sha256 = "<download and calculate it manually>",
    strip_prefix = "rules_go-a390e7f7eac912f6e67dc54acf67aa974d05f9c3",
    url = "https://github.com/bazelbuild/rules_go/archive/a390e7f7eac912f6e67dc54acf67aa974d05f9c3.zip",
)
```

**Comment by jayconrod** [comment]
Note that we have our own [`git_repository`](https://github.com/bazelbuild/bazel-gazelle/blob/master/repository.rst#git_repository) and [`http_archive`](https://github.com/bazelbuild/bazel-gazelle/blob/master/repository.rst#http_archive) which are more friendly for overlaying multiple build files. The native versions are fine for bootstrapping, unless they're actually going to be removed. So we may not want to change the setup instructions unless the Skylark versions provide the same functionality. Personally, I don't find the patch functionality easy to use.

**Comment by drigz** [comment]
Looks like this was fixed by #1516.
